### PR TITLE
[#6] [UI] As a user, I can see total amount voucher, and member in the cart section

### LINKED ISSUE #6: [UI] As a user, I can see total amount voucher,  and member in the cart section
## Why

When the user clicked on the sign-in button. It should navigate the user to MainScreen and they can see the total amount and voucher code in the cart.

## Acceptance Criteria

- Display the total in the cart section:
  - Display the label `Total`.
  - Display the total price.
  Screen Shot 2566-07-03 at 17 07 31
      
- Display the voucher code in the cart section:
  - Display the text field for entering the `voucher code`.
  - Display the hint of `text field` = `Enter voucher code`.
  - Display the `APPLY` button.
  - Display the scan button.
     Screen Shot 2566-07-03 at 17 28 16

- Display the voucher code in the cart section:
  - Display the text field for entering the `membership no`.
  - Display the hint of `text field` = `Enter membership no.`.
  - Display the scan button.
    Screen Shot 2566-07-03 at 17 28 26

- The sub-sections `Payment Method` in this story will be left as blank spaces. It will be handled in separate tickets.  

## Design

Screen Shot 2566-07-03 at 16 38 58

## Resources

https://www.figma.com/file/QQJEM3O8LmntPcU8UrpXOU/POS-iOS-application?type=design&node-id=338-217&mode=design&t=DXWd1nPcwzyjKp0f-4

### PR DESCRIPTION
**Pull request by kaungkhantsoe**
- Close #6 

## What happened 👀

When the user clicked on the sign-in button. It should navigate the user to MainScreen and they can see the total amount and voucher code in the cart.

## Insight 📝

N/A

## Proof Of Work 📹

Screenshot 2023-07-10 at 7 08 48 PM

### CODE REVIEW

File: lib/screen/main/main_screen.dart [hunk 1]
```diff
@@ -25,92 +26,158 @@ class _MainScreenState extends ConsumerState<MainScreen> {
...
+                  color: ColorName.whiteSmokeColor,
+                  borderRadius:
+                      BorderRadius.all(Radius.circular(Dimens.borderRadius10)),
+                ),
+                child: Padding(
+                  padding: const EdgeInsets.all(Dimens.space20),
+                  child: Row(
+                    children: [
+                      Expanded(
+                        flex: 1,
```
Comment (Wadeewee) [inline lib/screen/main/main_screen.dart:139]:
```suggestion
                       
```

File: lib/screen/main/code_scanner.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  const CodeScanner(
+      {Key? key,
+      required VoidCallback onClickScan,
+      required VoidCallback onClickApply,
+      required String placeHolderString})
+      : _onClickScan = onClickScan,
+        _onClickApply = onClickApply,
+        _placeHolderString = placeHolderString,
+        super(key: key);
```
Comment (Wadeewee) [inline lib/screen/main/code_scanner.dart:21]: Optional: We can let them public for this case.

```suggestion
  final VoidCallback onClickScan;
  final VoidCallback onClickApply;
  final String placeHolderString;

  const CodeScanner({
    Key? key,
    required this.onClickScan,
    required this.onClickApply,
    required this.placeHolderString,
  }) : super(key: key);
```

**Review by Wadeewee — COMMENTED** [review]
Just a small concern for the rest LGTM 🙌

**Comment by kaungkhantsoe** [comment]
@Wadeewee Updated in https://github.com/nimblehq/showcase-flutter-build-web-desktop/pull/18/commits/03f98dc91827c66d90a24a4e74f29b3e62fcc544

**Review by Wadeewee — APPROVED** [review]

**Pull request merged**